src/main/scala/Main.scala

```diff
@@ -17,26 +17,104 @@
 
 package com.trifectalabs.myriad
 
-import pso.{PSOConfiguration, PSOSystemFactory, PSOExecutor}
+import pso.{PSOConfiguration, PSOSystemFactory, PSOExecutor,
+  ParticleTopology, TerminationCriteria}
+import SampleFunctions._
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
 
 object Main extends App {
-  def obj(sol: List[Double]): Double = {
-    sol.map(x => x*x).sum
+  def iter(particles: List[List[Double]], velocities: List[List[Double]],
```

Comment:
This is a library, so there should be no entry point. I get that you're just testing with this, but keep in mind that we'll need to remove this before cutting a release.